### DISCUSSION TITLE
Apikey expiration

### DISCUSSION THREAD

**Discussion by Nattalli** [supabase/supabase#10685] [Questions] [answered]
How long `Apikey` is available for the project? I thought it was stored for a year, but is it safe? Maybe I wasn't looking well enough, but I couldn't find a setting for `apikey` lifetime anywhere. Is there any life expectancy at all? Is it eternal?... I want to use the Supabase as a database but I don't plan to implement the front-end with javascript (I will use a constructor)

**Comment by GaryAustin1** [comment, marked as answer]
You can use jwt.io to decode tokens.  Looks like 10 years is the token expiration.
